anaUltraScurve.py

```diff
@@ -0,0 +1,355 @@
+#!/bin/env python
+import os
+import numpy as np
+from optparse import OptionParser
+from array import array
+from fitScanData import *
+from channelMaps import *
+from PanChannelMaps import *
+from gempython.utils.nesteddict import nesteddict as ndict
+
+#parser = OptionParser()
+from anaoptions import parser
+
+parser.add_option("-b", "--drawbad", action="store_true", dest="drawbad",
+                  help="Draw fit overlays for Chi2 > 10000", metavar="drawbad")
+parser.add_option("-f", "--fit", action="store_true", dest="SaveFile",
+                  help="Save the Fit values to Root file", metavar="SaveFile")
+parser.add_option("--IsTrimmed", action="store_true", dest="IsTrimmed",
+                  help="If the data is from a trimmed scan, plot the value it tried aligning to", metavar="IsTrimmed")
+
+
+parser.set_defaults(outfilename="SCurveData.root")
+
+(options, args) = parser.parse_args()
+filename = options.filename[:-5]
+os.system("mkdir " + filename)
+
+print filename
+outfilename = options.outfilename
+
+vToQb = -0.8
+vToQm = 0.05
+
+import ROOT as r
+
+r.gROOT.SetBatch(True)
+r.gStyle.SetOptStat(1111111)
+GEBtype = options.GEBtype
+inF = r.TFile(filename+'.root')
+
+if options.SaveFile:
+    outF = r.TFile(filename+'/'+outfilename, 'recreate')
+    myT = r.TTree('scurveFitTree','Tree Holding FitData')
+    pass
+#Build the channel to strip mapping from the text file
+#chToStrLUT = []
+#strToChLUT = []
+lookup_table = []
+pan_lookup = []
+for vfat in range(0,24):
+    #chToStrLUT.append([])
+    #strToChLUT.append([])
+    lookup_table.append([])
+    pan_lookup.append([])
+    for channel in range(0,128):
+        #chToStrLUT[vfat].append(0)
+        #strToChLUT[vfat].append(0)
+        lookup_table[vfat].append(0)
+        pan_lookup[vfat].append(0)
+        pass
+    pass
+
+buildHome = os.environ.get('BUILD_HOME')
+
+if GEBtype == 'long':
+    #intext = open(buildHome+'/vfatqc-python-scripts/macros/longChannelMap.txt', 'r')
+    intext = open(buildHome+'/gem-plotting-tools/setup/longChannelMap.txt', 'r')
+    pass
+if GEBtype == 'short':
+    #intext = open(buildHome+'/vfatqc-python-scripts/macros/shortChannelMap.txt', 'r')
+    intext = open(buildHome+'/gem-plotting-tools/setup/shortChannelMap.txt', 'r')
+    pass
+
+for i, line in enumerate(intext):
+    if i == 0: continue
+    mapping = line.rsplit('\t')
+    #chToStrLUT[int(mapping[0])][int(mapping[2]) - 1] = int(mapping[1])
```

Comment:
can we please change the name of lookup_table to chToStrLUT because it is more descriptive. lookup_table is a terrible variable name. I might as well call all my integers int1 int2 int3...